### DISCUSSION TITLE
auto-reconnect for radio/rotator

### DISCUSSION THREAD

**Discussion by oh2gba** [foldynl/QLog#486] [Ideas]
Good day,

I’d like to request an auto-reconnect feature for connected radios/rotors.

Use case: I usually leave QLog running while the radio is off (e.g., when connected to a cluster awaiting a rare DX). The connection loss dialog that pops up is somewhat disruptive. It would be ideal if QLog could automatically reconnect when the radio is turned back on.

Thank you 73

**Comment by foldynl** [comment]
since version 0.39, QLog is tolerant to some Hamlib errors and tries to ignore them. Unfortunately, it's not your case. To be honest, the auto-reconnect function doesn't seem like a good solution to me. How do you distinguish cases when you are just "waiting" for a new connection and you really lost a connection to your rig? It doesn't make sense to me to show in QLog that the rig is connected when it is turned off.

**Comment by oh2gba** [comment]
Some design work is indeed needed. What if we add an 'Auto Reconnect' option in the 'Equipment' menu? This would allow polling to continue indefinitely instead of triggering an alert pop-up. Keep the existing functionality in the Rig/Rotator widget as it is now, with the green/red colors, polling would continue, but 'no data' from the radio would turn the color to red.

And now when I think about it, the "Keep Options" has always felt a bit weird to me. Not entirely  sure what the use case for it is. Keeping things simple, consider removing that option?

**Comment by foldynl** [comment]
The "Keep Option" is used to maintain the Rig/Rot state (connected/disconnected) when you restart the QLog. 

I find your auto-connect feature somewhat user-unfriendly, especially when the Rig remains disconnected without any notification. Users might not notice subtle indicators, like color changes, which could help them identify the Rig's state. More importantly, how is the user informed in cases when a network connection or cable is disconnected? This approach could be improved. While auto-connect is a good concept, it's crucial that the user is informed when the Rig totally unavailable. Please provide an explanation and suggest a solution for improving this feature.

**Comment by oh2gba** [comment]
Good feedback! Currently, if 'Connect Rig' isn’t enabled when starting QLog, the disconnected indication is already quite subtle. Enhancing this—such as using a red/blinking VFO—could improve clarity. To avoid annoyance without a connected radio, this could be toggled off via the Equipment menu.

A simple approach might be to indicate the user when no data is available from the radio, regardless of the reason (e.g., radio off or technical issue).

I'm still not totally sure about the "Keep Option", why would you not want to keep the setting. 

Hopefully, QLog development continues. Moving the auto-reconnect feature to the radio settings could help keep the menu structure simple, which I’d personally prefer. However it will probably need to be a global setting, not per radio. As this will otherwise complicate the UI quite a bit.

**Comment by oh2gba** [comment]
A simpler way is to always poll the radio, with no option to turn polling off. Make the connection indication slightly clearer by some usability testing to find the best visibility. If users want to stop polling the radio(s), they can turn it off in the 'Equipment' menu. Remove the 'Keep Option' setting; just remember the last state. For multiple radios, the Rig widget dropdown will show which radio is connected by color. This change keeps the UI simple, without breaking anything. 

This will slightly complicate the setup of new radio's. This may be improved, but I'd separate the setup into a new feature request.

**Comment by foldynl** [comment]
hmm...good suggestion. But how do you inform the user that, for example, he entered the wrong IP address or that he is not connected to the Internet? This needs to be responded to immediately. Unfortunately, it is not possible to distinguish whether you just turned off the rig or you're not connected to the internal. Sorry, but I will not implement Rig "auto-reconnect" in the near future, because there is more questions.

BTW: Auto-reconnect was implemented, for example, in DXC. There is no problem if your connection drops for a while and you don't receive information. But if you are running QSO and the frequency is not set correctly due rig is "silently" disconnected and auto-reconnect is performing, then it is a big problem. You definitely wouldn't want that.

By the way, QLog does not support multiple connected rigs.
